GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -217,9 +218,9 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
 
             string previousCommit = null;
 
-            // Use the shallow file to find a recent commit to diff against to try and reduce the number of SHAs to check
-            DiffHelper blobEnumerator = new DiffHelper(this.Tracer, this.Enlistment, this.FileList, this.FolderList);
-            if (File.Exists(shallowFile))
+            // Use the shallow file to find a recent commit to diff against to try and reduce the number of SHAs to check.

```

Comment:
I think all of this logic of looking for the shallow file only applies to fastfetch, meaning you could move this into a method of `CheckoutPrefetcher` that overrides an empty virtual method in this class. With that approach, you can keep the `force` logic completely contained within `CheckoutPrefetcher` and not even pass that flag to its base constructor.